### DISCUSSION TITLE
Public Key generated from mnemonic is not as expected for EDDSA Coin

### DISCUSSION THREAD

**Discussion by ghost** [CoolBitX-Technology/coolwallet-js-sdk#398] [Q&A] [answered]
- Description
We are trying to integrate Algorand. The public Key returned by the function is not as expected. COIN_TYPE = '8000011b'
```
export default class ALGO extends COIN.EDDSACoin implements COIN.Coin {
    constructor() {
        super(params.COIN_TYPE);
    }

    async getAddress(transport: Transport, appPrivateKey: string, appId: string): Promise<string> {
        const publicKey = await this.getPublicKey(transport, appPrivateKey, appId, false);
        if (!publicKey) {
            throw new ERROR.SDKError(this.getAddress.name, 'public key is undefined');
        }
        console.log("This is the public key = ", publicKey)
        return pubKeyToAddress(publicKey);
    }
```
 
- Expected Behavior
Expected Public Key = 79bda01f694d1469b14611e22f4606e58288193ccba3c095e774775c6c2ad656

- Current Behaviour
isSLIP0010 = True, Public Key = 1798676a039c096a7396b9ac1672b7ec05008f55b1930d940ae8d573a794f7a2
isSLIP0010 = False, Public Key = ac13ae847e20c1162160b7d98c47c8c84beaee57f991dc48cd5a6f97ce756562

**Comment by hank121314** [comment, marked as answer]
Could you provide the bip44 path of `Expected Public Key`?

**Reply by ghost** [reply]
Derivation Path : 
```
m/44'/283'/0'/0/0
```

**Reply by hank121314** [reply]
Please also provide the mnemonic thanks!

**Reply by ghost** [reply]
Mnemonic used : cattle siren cabbage issue better street gap outdoor notable clarify there six

**Reply by hank121314** [reply]
The bip44 path is incorrect it should be `m/44'/283'/0'/0'/0'`.

**Reply by ghost** [reply]
Where do I make the change to get the Expected Public Key. I'm only configuring COIN_TYPE = '8000011b' and extending the class EDDSACoin. I'm not configuring the derivation path anywhere. The public key seems to be generated from this inputs. Am I missing something ?

**Reply by hank121314** [reply]
If the result of `this.getPublicKey` is not an expected result you can override its behavior.
